lib/src/parser.dart

```diff
@@ -300,3 +310,47 @@ class Parser {
         '"$value" is not an allowed value for option "${option.name}".');
   }
 }
+
+/// A `List` which uses another `List` as its underlying data structure, but
+/// with the elements in reverse.
+///
+/// The makes `removeAt(0)` much faster, which is critical for parser
+/// performance.
+///
+/// It would be better to use a `Queue`, but `List` was already part of the
```

Comment:
Note that package:collection does have a `QueueList` - likely not worth picking up the dependency though

Reply:
Ah, along with a comment that it's basically a copy from dart:collection :/ ... that's annoying. I don't think I want to create a third copy to avoid the dep :)